### ISSUE TITLE
Error when importing the CEF Python 47 module: undefined symbol: _ZN3cef7logging10LogMessageC1EPKcii

### ISSUE THREAD

**Issue by cztomczak** [cztomczak/cefpython#230]
Updated to CEF 47 in commit bc3b9aa - it builds fine, but there is this error when running example:
```
Traceback (most recent call last):
  File "wxpython.py", line 19, in <module>
    import cefpython_py27 as cefpython
ImportError: /home/czarek/github/cefpython/src/linux/binaries_64bit/cefpython_py27.so: undefined symbol: _ZN3cef7logging10LogMessageC1EPKcii
```

Reported with details on the CEF Forum:
www.magpcss.org/ceforum/viewtopic.php?f=6&t=14269&p=31224

@allestuetsmerweh had the same issue when building idrissha's fork:
https://github.com/cztomczak/cefpython/issues/195#issuecomment-182472487

@sidreesshah Did you encounter the same issue when building newer CEF and if so how did you fix this?

**Comment by cztomczak** [comment]
This can be fixed by changing the order of libraries during linking in linux/setup/setup.py:
```
     libraries=[
+        'cefpythonapp',
         'cef_dll_wrapper',
         'client_handler',
-        'cefpythonapp',
         'cpp_utils',
         'gtk-x11-2.0',
```

**Comment by cztomczak** [comment]
Fixed in 03e9515da0bf90073962e813f669ba8d437d2138

**Comment by cztomczak** [comment]
Previous commit didn't fix the issue, it still occured on some systems. Fixed again by adding -flto and a few other flags in commit 7d47bfe29740fa88eaa04c6c3853a7babe99c201.

New flags:
```
    extra_compile_args=['-flto', '-fdata-sections', '-ffunction-sections',
                        '-std=gnu++11'],
    extra_link_args=['-flto', '-Wl,--gc-sections'],
```

More details on the flags here: https://stackoverflow.com/questions/6687630/
